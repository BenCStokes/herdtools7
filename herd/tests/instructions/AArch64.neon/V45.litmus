AArch64 V45
(* Tests STP - store pair, post-indexed *)
{
uint64_t x[2];
0:X0=x; 0:X1=y;
}

 P0                      ;
 MOVI D0, #254           ;
 MOVI D1, #255           ;
 STP D0, D1, [X0], #-8   ;
 LDR D2, [X0, #16]       ;
 STR D2, [X1]            ;
 forall (x=254 /\ y=255 /\ 0:X0!=x)
