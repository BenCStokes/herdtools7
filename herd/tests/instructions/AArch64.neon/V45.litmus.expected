Test V45 Required
States 1
0:X0=x+-8; x={254,255}; y=255;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (x=254 /\ y=255 /\ not (0:X0=x))
Observation V45 Always 1 0
Hash=67a095552f8fca01bbb8e74c11bca4ce

